src/api.jl

```diff
@@ -0,0 +1,188 @@
+"""
+    Julja2ParsedTemplate
+
+Represents a parsed template with executable generator code.
+
+## Fields:
+- `generator_code::Expr`: The expression that generates the code for the template.
+"""
+struct Julja2ParsedTemplate
+generator_code::Expr
+end
+
+"""
+    BlockBorders
+
+This structure contains borders of block.
+
+## Fields:
+- `left::String`: The left border string.
+- `left_length::Int`: The length of the left border.
+- `right::String`: The right border string.
+- `right_length::Int`: The length of the right border.
+"""
+mutable struct BlockBorders
+    left::String
+    left_length::Int
+    right::String
+    right_length::Int
+
+    function BlockBorders(left::String, right::String)
+        # Get UInt8 string length because unicode characters may be more than one byte.
+        new(
+            left,
+            length(transcode(UInt8, left)),
+            right,
+            length(transcode(UInt8, right))
+        )
+    end
+end
+
+"""
+    Julja2Template(template::String; var_borders::BlockBorders=BlockBorders("{{", "}}"), 
+    operator_borders::BlockBorders=BlockBorders("{%", "%}"), templates_dir::String="./", 
+    ignore_missing_vars=true, allow_filename_change_dir=false) -> Julja2ParsedTemplate
+
+Creates an executable template to create result string.
+
+## Required parameters:
+- `template::String`: The template string to parse.
+
+## Optional parameters:
+- `var_borders::BlockBorders`: Borders for variable blocks (default: `BlockBorders("{{", "}}")`).
+- `operator_borders::BlockBorders`: Borders for operator blocks (default: `BlockBorders("{%", "%}")`).
+- `templates_dir::String`: Path (relative or absolute) to files that will be included with `{% include %}` operator (default: `"./"`).
+- `ignore_missing_vars::Bool`: If true, raises an error if a variable in `{{ variable_block }}` is missing (default: `true`).
+- `allow_filename_change_dir::Bool`: Recommended to set to true as it could be a potential vulnerability. It forbids "/" in file names to prevent directory changes in Unix systems (default: `false`).
+
+## Returns:
+- `Julja2ParsedTemplate`: The parsed template object.
+"""
+function Julja2Template(template::String;
+                var_borders::BlockBorders=BlockBorders("{{", "}}"),
+                operator_borders::BlockBorders=BlockBorders("{%", "%}"),
+                templates_dir::String="./",
+                ignore_missing_vars=true,
+                allow_filename_change_dir=false)::Julja2ParsedTemplate
+
+    template_dir_path = _format_template_dir_path(templates_dir)
+    template_options = TemplateOptions(
+        var_borders, operator_borders, template_dir_path,
+        ignore_missing_vars, allow_filename_change_dir)
+    _validate_options(template_options)
```

Comment:
Sometimes it can be difficult, but it is better to try to fit into at least two words in the name of functions and variables
Also, if the algorithm makes it clear what the variable is for, then some of the clarifying words can be omitted (`template_options`
-> `options = TemplateOptions(...)`, `ignore_missing_vars` -> `ignore_missing`, ...)